test/functional/ui/messages_spec.lua

```diff
@@ -24,6 +24,31 @@ local is_os = t.is_os
 local fn = n.fn
 local skip = t.skip
 
+clear()
+local hlmap = {
+  '@character',
+  '@character.special',
+  'DiffAdd',
+  'DiffChange',
+  'Directory',
+  'Error',
+  'ErrorMsg',
+  'LineNr',
+  'ModeMsg',
+  'MoreMsg',
+  'MsgArea',
+  'Question',
+  'Search',
+  'Special',
+  'SpecialKey',
+  'StderrMsg',
+  'Title',
+  'WarningMsg',
+}
+for _, hl in ipairs(hlmap) do
+  hlmap[hl] = api.nvim_get_hl_id_by_name(hl)
```

Comment:
Nice. Future idea: perhaps we should put this in screen.lua and use it in more places

Reply:
Yeah it would be nice if `_print_snapshot()` spits out the highlight group instead of the ID, otherwise future tests not using this could run into the same problem. I'll see if I can make that work.